provisioner/ansible/provisioner_test.go

```diff
@@ -741,3 +742,74 @@ func TestUseProxy(t *testing.T) {
 		os.Remove(p.config.Command)
 	}
 }
+
+func TestProvisionerPrepare_WinRMSSL(t *testing.T) {
+	testcases := []struct {
+		name            string
+		winRMUseSSL     bool
+		extraArgs       []string
+		expectedArgs    []string
+		expectSchemeArg bool
+	}{
+		{
+			name:            "winrm_use_ssl set to false, should specify -e ansible_winrm_scheme",
+			winRMUseSSL:     false,
+			extraArgs:       []string{},
+			expectedArgs:    []string{},
+			expectSchemeArg: true,
+		},
+		{
+			name:            "winrm_use_ssl set to false, option defined, should not specify -e ansible_winrm_scheme twice",
+			winRMUseSSL:     false,
+			extraArgs:       []string{"-e", "ansible_winrm_scheme=http"},
+			expectedArgs:    []string{},
+			expectSchemeArg: true,
+		},
+		{
+			name:            "winrm_use_ssl set to true, option defined, should not specify -e ansible_winrm_scheme twice",
+			winRMUseSSL:     true,
+			extraArgs:       []string{"-e", "ansible_winrm_scheme=http"},
+			expectedArgs:    []string{},
+			expectSchemeArg: true,
+		},
+		{
+			name:            "winrm_use_ssl set to true, option defined, should not specify -e ansible_winrm_scheme at all",
```

Comment:
I believe if the winrm_use_ssl is true no extra args should be added right? 
```suggestion
			name:            "winrm_use_ssl set to true, should not specify -e ansible_winrm_scheme at all",
```

Reply:
This is a fair point, I have not defined a test without the arg defined, but I can add this for completeness.

Reply:
Definitely a good call, since the option should default to true, I had to rework it to use a Trilean to disambiguate between a proper `false` and the default `false` value.
Tests are also defined to show for it.